src/layouts/components/Editor/MenuItem.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from "@chakra-ui/react"
+import { IconButton } from "@opengovsg/design-system-react"
+import { MouseEventHandler } from "react"
+import remixiconUrl from "remixicon/fonts/remixicon.symbol.svg"
+
+interface MenuItemProps {
+  icon?: string
+  title?: string
+  action?: MouseEventHandler<HTMLButtonElement>
+  isActive?: null | (() => boolean)
+}
+
+export default ({ icon, title, action, isActive = null }: MenuItemProps) => (
+  <IconButton
+    // variant="unstyled"
+    type="button"
+    className="menu-item"
+    _hover={{ bg: "gray.100" }}
+    _active={{ bg: "gray.200" }}
+    onClick={action}
+    title={title}
+    bgColor={isActive && isActive() ? "gray.200" : "transparent"}
```

Comment:
could we rename the isActives to represent what each type of `isActive` here refers to?

Reply:
no, because we require a function call as otherwise this will be static.